cdap-ui/app/services/cask-angular-socket-datasource/datasource.js

```diff
@@ -172,7 +172,7 @@ var socketDataSource = angular.module(PKG.name+'.services');
 
       function startClientPoll(resourceId, bindings, interval) {
         const intervalTimer = setTimeout(() => {
-          const resource = bindings[resourceId].resource;
+          const resource = bindings[resourceId]? bindings[resourceId].resource : undefined;
```

Comment:
Should we also check the nullness of `bindings`?

const resource = bindings && bindings[resourceId]? bindings[resourceId].resource : undefined;

Reply:
Good question - we check for the nullness of bindings before we call startClientPoll, so we don't need to do it here. Also, I don't think bindings can ever be null because we initialize it as an empty object anyway.